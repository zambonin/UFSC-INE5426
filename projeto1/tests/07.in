bool fun f ()
bool fun f () {
ret false
}
if f() {
  int a = 0
  int fun f2 ( int x ) {
          int a
         a = x + 1
         ret a
  }
  a = f2 ( a)
}

int fun fibo ( int x, bool b )

int fun fibo ( int x, bool b ) {
	int ans
	if x < 2
	then {
		ans = x
	} else {
		ans = fibo(x-1, b) + fibo(x-2, b)
	}
	ret ans
}

int a
a = fibo (10, true)

