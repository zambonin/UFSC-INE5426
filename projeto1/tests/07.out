int fun: f (params: )
  ret false
if: f[0 params]
then:
  int var: a = 0
  int fun: f2 (params: int x)
    int var: a
    = a + x 1
    ret a
  = a f2[1 params] a
int fun: fibo (params: int x, bool b)
  int var: ans
  if: < x 2
  then:
    = ans x
  else:
    = ans + fibo[2 params] - x 1 b fibo[2 params] - x 2 b
  ret ans
int var: a
= a fibo[2 params] 10 true

